src/repr/src/scalar.rs

```diff
@@ -3767,6 +3796,145 @@ pub fn arb_datum() -> BoxedStrategy<PropDatum> {
     .boxed()
 }
 
+/// Generates an arbitrary [`PropDatum`] for the provided [`ColumnType`].
+pub fn arb_datum_for_column(column_type: &ColumnType) -> impl Strategy<Value = PropDatum> {
+    let strat = arb_datum_for_scalar(&column_type.scalar_type);
+
+    if column_type.nullable {
+        Union::new_weighted(vec![(1, Just(PropDatum::Null).boxed()), (5, strat.boxed())]).boxed()
+    } else {
+        strat.boxed()
+    }
+}
+
+/// Generates an arbitrary [`PropDatum`] for the provided [`ScalarType`].
+pub fn arb_datum_for_scalar(scalar_type: &ScalarType) -> impl Strategy<Value = PropDatum> {
+    match scalar_type {
+        ScalarType::Bool => any::<bool>().prop_map(PropDatum::Bool).boxed(),
+        ScalarType::Int16 => any::<i16>().prop_map(PropDatum::Int16).boxed(),
+        ScalarType::Int32 => any::<i32>().prop_map(PropDatum::Int32).boxed(),
+        ScalarType::Int64 => any::<i64>().prop_map(PropDatum::Int64).boxed(),
+        ScalarType::PgLegacyChar => any::<u8>().prop_map(PropDatum::UInt8).boxed(),
+        ScalarType::UInt16 => any::<u16>().prop_map(PropDatum::UInt16).boxed(),
+        ScalarType::UInt32
+        | ScalarType::Oid
+        | ScalarType::RegClass
+        | ScalarType::RegProc
+        | ScalarType::RegType => any::<u32>().prop_map(PropDatum::UInt32).boxed(),
+        ScalarType::UInt64 => any::<u64>().prop_map(PropDatum::UInt64).boxed(),
+        ScalarType::Float32 => any::<f32>().prop_map(PropDatum::Float32).boxed(),
+        ScalarType::Float64 => any::<f64>().prop_map(PropDatum::Float64).boxed(),
+        ScalarType::Numeric { .. } => arb_numeric().prop_map(PropDatum::Numeric).boxed(),
+        ScalarType::String
+        | ScalarType::PgLegacyName
+        | ScalarType::Char { length: None }
+        | ScalarType::VarChar { max_length: None } => ".*".prop_map(PropDatum::String).boxed(),
+        ScalarType::Char {
+            length: Some(length),
+        } => {
+            // Constrain the length otherwise we try to generate massive strings.
```

Comment:
This means we'd generate data that doesn't actually match the type, right? WDYT about capping the length of the `ScalarType::Char`s we generate instead?

Reply:
Doh! yeah that's definitely the way to go, updated!

Also updated the strategy for generating `Char`s to be more realistic